src/App.tsx

```diff
@@ -1,61 +1,104 @@
+import React, { useState } from 'react';
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import { TodosWithUsers } from './react-app-env';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+export const todosWithUsers: TodosWithUsers[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: usersFromServer.find(user => user.id === todo.userId) || null,
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<TodosWithUsers[]>(todosWithUsers);
+  const [newTitle, setNewTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [emptyTitleError, setEmptyTitleError] = useState(false);
+  const [emptyUserError, setemptyUserError] = useState(false);
+
+  const getBigestId = () => {
+    return Math.max(...(todos.map(todo => todo.id))) + 1;
+  };
+
+  const resetForm = () => {
+    setNewTitle('');
+    setSelectedUser(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setEmptyTitleError(!emptyTitleError);
+    setemptyUserError(!emptyUserError);
```

Comment:
Add if statement here to check if you have errors - stop code execution if you have errors here.